### PR TITLE
docs: Replace Apache license headers with MIT

### LINKED ISSUE #399: License discrepancy in source files vs LICENSE file
It looks like the LICENSE file for this library is MIT, but the source files contain Apache licenses in JS comments. Is this discrepancy intentional?

If not, which one is the appropriate license for the library?

### PR DESCRIPTION
**Pull request by danielduhh**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/gcs-resumable-upload/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #399  🦕

### CODE REVIEW

File: .kokoro/docs.sh [hunk 1]
```diff
@@ -2,17 +2,9 @@
 
 # Copyright 2018 Google LLC
 #
-# Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (JustinBeckwith) [inline .kokoro/docs.sh:5]: Oh my - so all of the files in `.kokoro`, `.prettierignore`, `.mocharc.js`, `.jsdoc.js` - those will continue to have an Apache license.  They are brought in as part of a set of templates that are consistent across all of our nodejs libraries.  None of these files are shipped with the product code in the `npm` package, so I don't think this is going to cause us any trouble.  We should revert these.

Comment (danielduhh) [reply .kokoro/docs.sh:5]: Ok no prob

Comment (danielduhh) [reply .kokoro/docs.sh:5]: Just calling out that we'll continue to have the discrepancies mentioned in #399 ...

**Comment by danielduhh** [comment]
fyi @JustinBeckwith , This will fail until this [header-check bot issue](https://github.com/googleapis/repo-automation-bots/pull/1304) is fixed and released

**Review by bcoe — APPROVED** [review]

**Pull request merged**